packages/python-packages/api-stub-generator/apistub/nodes/_function_node.py

```diff
@@ -158,17 +173,22 @@ def _parse_docstring(self):
                     self.args.append(kw_arg)
 
             # API must have **kwargs. Flag it as an error if it is missing for public API
-            if not kwargs and not self.name.startswith("_"):
+            if not kwargs and is_kwarg_mandatory(self.name):
                 self.errors.append("Keyword arg (**kwargs) is missing in method {}".format(self.name))
 
 
     def _parse_typehint(self):
+
+        # Skip parsing typehint if typehint is not expected for e.g dunder or async methods
+        if not is_typehint_mandatory(self.name) or self.is_async:
```

Comment:
In the case of async we should be checking the type hints in the method signature itself - unless that already lives elsewhere? Or that could be added later if the intention is just to remove the false negative for now.

Reply:
Parser first looks for type hint in signature which is returned by inspect module. So for any async methods, signature will have type name if typehint is part of it. If typehint is missing in signature then parser flag those params as missing type error.